eo-maven-plugin/src/main/java/org/eolang/maven/PullMojo.java

```diff
@@ -152,17 +144,23 @@ public void exec() throws IOException {
                 this.forceUpdate()
             );
         }
-        if (!tojos.isEmpty()) {
-            for (final Tojo tojo : tojos) {
-                tojo.set(
-                    AssembleMojo.ATTR_EO,
-                    this.pull(tojo.get(Tojos.KEY)).toAbsolutePath().toString()
-                );
-                tojo.set(
-                    AssembleMojo.ATTR_HASH,
-                    new ChNarrow(hash).value()
-                );
+        for (final Tojo tojo : tojos) {
+            tojo.set(
+                AssembleMojo.ATTR_EO,
+                this.pull(tojo.get(Tojos.KEY)).toAbsolutePath().toString()
+            );
+            tojo.set(
+                AssembleMojo.ATTR_HASH,
+                new ChNarrow(hash).value()
+            );
+        }
+        if (tojos.isEmpty()) {
+            if (this.scopedTojos().select(row -> true).isEmpty()) {
+                Logger.warn(this, "Nothing to pull, since there are no foreign programs");
+            } else {
+                Logger.info(this, "Nothing to pull, all programs pulled already");
```

Comment:
@MikhailLipanin Is it reachable state? Please, provide some tests to prove it.

Reply:
@volodya-lombrozo, Logging was returned back (as it was earlier)